### PR TITLE
Add Llama 3.1 Pruning and Distillation Tutorial

### PR DESCRIPTION
**Pull request by gvenkatakris**
Add Llama 3.1 pruning and distillation tutorial to NeMo tutorials

### CODE REVIEW

File: tutorials/llm/llama-3/pruning-distillation/README.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+`NVIDIA NeMo Framework <https://docs.nvidia.com/nemo-framework/user-guide/latest/overview.html>`_ provides tools to perform teacher finetuning, pruning and distillation on Llama 3.1 to fit your use case.
+
+Objectives
+----------
+
+This tutorial shows how to perform depth-pruning, teacher finetuning and distillation on **Llama 3.1 8B Instruct** using the `WikiText-103-v1 <https://huggingface.co/datasets/Salesforce/wikitext/viewer/wikitext-103-v1>`_ dataset with NeMo Framework. The `WikiText-103-v1 <https://huggingface.co/datasets/Salesforce/wikitext/viewer/wikitext-103-v1>`_ language modeling dataset is a collection of over 100 million tokens extracted from the set of verified Good and Featured articles on Wikipedia. For this demonstration, we will perform a light finetuning procedure on the ``Meta Llama 3.1 8B Instruct`` teacher model to generate a finetuned teacher model ``megatron_llama_ft.nemo``. This finetuned teacher model is then depth-pruned to create a trimmed model ``4b_trimmed_model.nemo``. These models will serve as a starting point for distillation to create a final distilled 4B model.
```
Comment (sharathts) [inline tutorials/llm/llama-3/pruning-distillation/README.rst:11]: "to generate a finetuned teacher model needed for optimal distillation"

**Review by sharathts — DISMISSED** [review]

**Review by sharathts — DISMISSED** [review]

File: tutorials/llm/llama-3/pruning-distillation/images/val_loss_distillation.png
Comment (ericharper) [inline tutorials/llm/llama-3/pruning-distillation/images/val_loss_distillation.png:1]: We don't allow any image data in .git. Please source the image from somewhere. You can add it as a release asset if needed.

Comment (gvenkatakris) [reply tutorials/llm/llama-3/pruning-distillation/images/val_loss_distillation.png:1]: Switched to using image URL. Rendering issue observed when inserting image in Jupyter Notebook. Image can be seen when the notebook is downloaded. Also added the image to README.

**Review by sharathts — APPROVED** [review]

**Comment by zhaoyang-star** [comment]
Thanks for this great feature! I note that the current impl only support TP. Will PP be supported in the roadmap?

**Pull request merged**